tests/gateways/test_repodata_gateway.py

```diff
@@ -175,13 +175,18 @@ def test_coverage_repodata_state(tmp_path):
     # now these should be loaded through RepodataCache instead.
 
     # assert invalid state is equal to no state
-    state = RepodataState(
-        tmp_path / "garbage.json",
-        tmp_path / f"garbage{CACHE_STATE_SUFFIX}",
-        "repodata.json",
+    state = RepodataCache(
+        tmp_path,
+        "garbage.json",
     )
-    state.cache_path_state.write_text("not json")
-    assert dict(state.load()) == {}
+    state.load_state().cache_path_state.write_text("{}")
+
+    assert state.load_state().data == {}
+
+    # assert writing something loads same
+    state.cache_path_state.write_text('{"jim":"bob"}')
```

Comment:
This part fails so far. Maybe the mtime is wrong on one of the files or we got a filename wrong.

Check coverage.